tests/components/tuya/test_config_flow.py

```diff
@@ -0,0 +1,147 @@
+"""Tests for the Tuya config flow."""
+import pytest
+from tuyaha.tuyaapi import TuyaAPIException, TuyaNetException
+
+from homeassistant import config_entries, data_entry_flow, setup
+from homeassistant.components.tuya.const import CONF_COUNTRYCODE, DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_PLATFORM, CONF_USERNAME
+
+from tests.async_mock import Mock, patch
+from tests.common import MockConfigEntry
+
+USERNAME = "myUsername"
+PASSWORD = "myPassword"
+COUNTRY_CODE = "1"
+TUYA_PLATFORM = "tuya"
+
+TUYA_USER_DATA = {
+    CONF_USERNAME: USERNAME,
+    CONF_PASSWORD: PASSWORD,
+    CONF_COUNTRYCODE: COUNTRY_CODE,
+    CONF_PLATFORM: TUYA_PLATFORM,
+}
+
+
+@pytest.fixture(name="tuya")
+def tuya_fixture() -> Mock:
+    """Patch libraries."""
+    with patch("homeassistant.components.tuya.config_flow.TuyaApi") as tuya:
+        yield tuya
+
+
+async def test_user(hass, tuya):
+    """Test user config."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)